include/ensmallen_bits/cmaes/cmaparameters.hpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file cmaesparametes.hpp
+ * @author John Hoang
+ * @author Marcus Edel 
+ * 
+ * Definition of the Covariance Matrix Adaptation Evolution Strategy as proposed
+ * by N. Hansen et al. in "Completely Derandomized Self-Adaptation in Evolution
+ * Strategies".
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_CMAES_CMA_PARAMETERS_HPP
+#define ENSMALLEN_CMAES_CMA_PARAMETERS_HPP
+
+#include "weight_init_policies/default_weight.hpp"
+#include "weight_init_policies/negative_weight.hpp"
+namespace ens{
+  class CMAparameters {
```

Comment:
I looked into the parameter class, and I think we can implement this using the policy design pattern. I would restructure this by implementing a `StandardCMAES` policy first, which implements the classic CMAES method. Following something simple as, https://github.com/mlpack/ensmallen/blob/master/include/ensmallen_bits/adam/adam_update.hpp with the additional methods to initialize the CMAES parameters and something like https://github.com/mlpack/ensmallen/blob/master/include/ensmallen_bits/sgd/sgd_impl.hpp as the base class.

Reply:
I can restructure the code, if you like, just let me know, should be straightforward.